realm/src/androidTest/java/io/realm/RealmInterprocessTest.java

```diff
@@ -0,0 +1,260 @@
+    // This handler run in newly created thread's Looper. When writing the test case, remember to call
+    // Looper.loop() to start handling message.
+    // Pass the first thing you want to run to the constructor which will be posted to the beginning
+    // of the message queue.
+    // Write the comments of the test case like this:
+    // A-Z means steps running from service process.
+    // 1-9xx means steps running from the main process.
+    @SuppressLint("HandlerLeak") // SuppressLint bug, doesn't work
+    private class InterprocessHandler extends Handler {
+        // Timeout Watchdog. In case the service crashed or expected response is not returned.
+        // It is very important to feed the dog after the expected message arrived.
+        private final int timeout = 5000;
+        private boolean timeoutFlag = true;
+        private Runnable timeoutRunnable = new Runnable() {
+            @Override
+            public void run() {
+                if (timeoutFlag) {
+                    assertTrue("Timeout happened", false);
+                } else {
+                    timeoutFlag = true;
+                    postDelayed(timeoutRunnable, timeout);
+                }
+            }
+        };
+
+        protected void clearTimeoutFlag() {
+            timeoutFlag = false;
+        }
+
+        protected void done() {
+            Looper.myLooper().quit();
+        }
+
+        public InterprocessHandler(Runnable startRunnable) {
+            super(Looper.myLooper());
+            receiverMessenger = new Messenger(this);
+            // To have the first step from main process run
+            post(startRunnable);
+            // Start watchdog
+            postDelayed(timeoutRunnable, timeout);
+        }
+
+        @Override
+        public void handleMessage(Message msg) {
+            Bundle bundle = msg.getData();
+            String error = bundle.getString(InterprocessService.BUNDLE_KEY_ERROR);
+            if (error != null) {
+                // Assert and show error from service process
+                assertTrue(error, false);
+            }
+        }
+    }
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+
+        Realm.deleteRealm(new RealmConfiguration.Builder(getContext()).build());
+
+        // Start the testing service
+        serviceLatch = new CountDownLatch(1);
+        Intent intent = new Intent(getContext(), InterprocessService.class);
+        getContext().bindService(intent, serviceConnection, Context.BIND_AUTO_CREATE);
+        // Wait until service started
+        assertTrue(serviceLatch.await(10, TimeUnit.SECONDS));
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        if (testRealm != null) {
+            testRealm.close();
+        }
+
+        getContext().unbindService(serviceConnection);
+        serviceMessenger = null;
+
+        // Wait until the service process terminated.
+        while (getServiceInfo() != null) {
```

Comment:
This looks a bit dangerous. I would add a counter and throw an exception after 10 tries or something